jetbrains-core/src/software/aws/toolkits/jetbrains/core/stack/Stack.kt

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.core.stack
+
+import com.intellij.notification.NotificationGroup
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.project.Project
+import com.intellij.ui.OnePixelSplitter
+import software.amazon.awssdk.services.cloudformation.model.StackStatus
+import javax.swing.SwingUtilities
+
+private const val STACK_TOOLWINDOW_ID = "AWS Stack"
+private const val UPDATE_STACK_STATUS_INTERVAL = 5000
+private const val REDRAW_ANIMATED_ICON_INTERVAL = 70
+private const val TREE_TABLE_INITIAL_PROPORTION = 0.25f
+/**
+ * Package entry point
+ * Opens stack UI ToolWindow. Window is registered on this call and unregistered when closed.
+ */
+fun openStack(project: Project, stackName: String) {
+    assert(SwingUtilities.isEventDispatchThread())
+    StackUI(project, stackName).start()
+}
+
+private class StackUI(
+    private val project: Project,
+    private val stackName: String
+) : UpdateListener {
+    private val toolWindow: ToolWindow
+    private val animator: IconAnimator
+    private val updater: Updater
+    private val notificationGroup: NotificationGroup
+
+    init {
+        val tree = TreeViewImpl(project, stackName)
+        animator = IconAnimator(REDRAW_ANIMATED_ICON_INTERVAL, tree)
+        val table = TableViewImpl { next -> print("Button clicked $next") }.apply { enableButtons(prev = false, next = true) }
```

Comment:
remove the print()

Reply:
It is temporary print, I will remove it with next commit for #472